File: Batch75/ValidAnagram/ValidAnagram.js
const ValidAnagram = (s,t) => {
  let hashMapS = {}
  for (let i = 0; i < s.length; i++) {
      hashMapS[s[i]] ? hashMapS[s[i]] += 1 : hashMapS[s[i]]  = 1
  }
  for (let i = 0; i < t.length; i++) {
      if(!hashMapS[t[i]] || s.length !== t.length) {
          return false
      } else {
          hashMapS[t[i]] -= 1
      }
  }
  return true
};

File: data-structures-py/Array-Hashmap/ValidAnagram.py
# Given two strings s and t, return true if t is an anagram of s, and false otherwise.
# An Anagram is a word or phrase formed by rearranging the letters of a different 
# word or phrase, typically using all the original letters exactly once.

class Solution(object):
  def validAnagram(self,s,t ) -> bool:
    hashmap = {}
    for x in s:
      if x in hashmap:
        hashmap[x] += 1
      else:
        hashmap[x] = 1

    for y in range(len(t)):
      if y in hashmap:
        hashmap[y] += 1
      else:
        return False

    return True

